django/template/defaulttags.py

```diff
@@ -1564,3 +1567,186 @@ def do_with(parser, token):
     nodelist = parser.parse(("endwith",))
     parser.delete_first_token()
     return WithNode(None, None, nodelist, extra_context=extra_context)
+
+
+class TemplateProxy:
+    """
+    A lightweight Template lookalike used for template partials.
+
+    Wraps nodelist as partial, in order to bind context.
+    """
+
+    def __init__(self, nodelist, origin, name):
+        self.nodelist = nodelist
+        self.origin = origin
+        self.name = name
+
+    def get_exception_info(self, exception, token):
+        template = self.origin.loader.get_template(self.origin.template_name)
+        return template.get_exception_info(exception, token)
+
+    def find_partial_source(self, full_source, partial_name):
+        result = ""
+        pos = 0
+        for m in partial_start_tag_re.finditer(full_source, pos):
+            sspos, sepos = m.span()
+            starter, name, inline = m.groups()
+            endm = partial_end_tag_re.search(full_source, sepos + 1)
+            assert endm, "End tag must be present"
+            espos, eepos = endm.span()
+            if name == partial_name:
+                # Include the full partial definition from opening to closing tag
+                result = full_source[sspos:eepos]
+                break
+            pos = eepos + 1
+        return result
+
+    @property
+    def source(self):
+        template = self.origin.loader.get_template(self.origin.template_name)
+        return self.find_partial_source(template.source, self.name)
+
+    def _render(self, context):
+        return self.nodelist.render(context)
+
+    def render(self, context):
+        "Display stage -- can be called many times"
```

Comment:
This is not following the documented style for docstrings, it should use triple double quotes, and ideally it would only be defined if the docstring adds valuable information not clear from the implementation.

Reply:
I removed it.  As it is not needed.